### ISSUE TITLE
Slow query on memberlist

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/SMF#7208]
#### Description

```
SELECT mem.id_member
FROM smf_members AS mem
   
WHERE mem.is_activated = 1
ORDER BY mem.real_name ASC
LIMIT 0, 30
```

id | select_type | table | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | SIMPLE | mem |   | ref | is_activated | is_activated | 1 | const | 287230 | 100.00 | Using index condition; Using filesort

### Steps to reproduce
1. 
2. 

### Environment (complete as necessary)
- Version/Git revision: N/A
- Database Type: Mysql
- Database Version: 5.7
- PHP Version: 7.4

### Additional information/references
https://www.simplemachines.org/community/index.php?topic=580082

Attaching this to 2.1.1, as I'm not sure if we can fix it for 2.1.0.  But we may want to fix for 2.1.0, as I suspect it may be a new index issue.

@albertlast See what you can do?

**Comment by albertlast** [comment]
I did look in my test env (pg),
look like that i create a test index very long time ago,
where the colum (is_activated, real_name) as own idx exists for the table smf_members.

So you could try to add this and see what mysql is doing with this.

**Comment by jdarwood007** [comment]
That seems to fix it.

The other sorts such as position, website also do this.

Sorting by membergroup
```
      SELECT mem.id_member
      FROM smf_members AS mem
         LEFT JOIN smf_membergroups AS mg ON (mg.id_group = CASE WHEN mem.id_group = 0 THEN mem.id_post_group ELSE mem.id_group END)
         
      WHERE mem.is_activated = 1
      ORDER BY mg.group_name is not null, mg.group_name ASC
      LIMIT 0, 30
   in /Sources/Memberlist.php line 353, which took 13.64280391 seconds at 0.25429106 into request.
```
id | select_type | table | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | SIMPLE | mem |   | ref | is_activated,custom_memberlist_sort | is_activated | 1 | const | 287243 | 100.00 | Using temporary; Using filesort
1 | SIMPLE | mg |   | eq_ref | PRIMARY | PRIMARY | 2 | func | 1 | 100.00 | Using where


Sorting by website
```
SELECT mem.id_member
FROM main_members AS mem
   
WHERE mem.is_activated = 1
ORDER BY mem.website_url != '', mem.website_url is not null, mem.website_url ASC
LIMIT 0, 30
in /Sources/Memberlist.php line 353, which took 3.75883389 seconds at 0.26961303 into request.
```
id | select_type | table | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | SIMPLE | mem |   | ref | is_activated,custom_memberlist_sort | is_activated | 1 | const | 287243 | 100.00 | Using index condition; Using filesort



Sorting by website seems we can fix with an index, but it seems wasteful to do.  sorting by group I'm not sure how we can optimize the query.  Maybe an index on is_activated and id_group would do that one.

**Comment by albertlast** [comment]
personaly i would drop the feature of being able to sort by website to be ownest,
i see the potention ob being ddos by this way and i see no need for this.

**Comment by Oldiesmann** [comment]
The ability to sort by website has been there since the beginning to my knowledge, and I've never heard of a DDOS because of it. I'm not sure how many people use that option, but removing something because of potential for DDOS (when nothing of the sort has been reported in the nearly 20 year history of SMF (1.0 Beta 1a was released in 2003)) seems a bit harsh

**Comment by albertlast** [comment]
a feature is living by his use and not by his existing,
so how often is it used?

**Comment by jdarwood007** [comment]
How about we fix the indexes, but prevent the sorting for website for guests?  Reduces the attack layer to signed in users and admins then can then better use tools to prevent attacks.

**Comment by Sesquipedalian** [comment]
If anyone wants to submit a fix soon, I'll consider it for 2.1.2, but I don't want to hold up 2.1.2 for this issue.

**Comment by albertlast** [comment]
#7415 and #7416 created for the mention from @jdarwood007
